web/src/lib/taxation/helpers.ts

```diff
@@ -0,0 +1,55 @@
+import { postTaxFilingsOnboarding } from "@/lib/api-client/apiClient";
+import { UpdateQueue } from "@/lib/types/types";
+import { LookupLegalStructureById } from "@businessnjgovnavigator/shared/legalStructure";
+import { ProfileData } from "@businessnjgovnavigator/shared/profileData";
+
+const displayBusinessName = (profileData: ProfileData): boolean => {
+  return LookupLegalStructureById(profileData.legalStructureId).elementsToDisplay.has("businessName");
+};
+
+const displayResponsibleOwnerName = (profileData: ProfileData): boolean => {
+  return LookupLegalStructureById(profileData.legalStructureId).elementsToDisplay.has("responsibleOwnerName");
+};
+
+interface Props {
+  updateQueue: UpdateQueue;
+  optionalProfileDataIfSeparateContext?: ProfileData;
+}
+// the aim of this fn is to hit postTaxFilingsOnboarding and staged update queue
+export const toBeNamed = async (props: Props): Promise<void> => {
+  const currentBusiness = props.updateQueue.currentBusiness();
+
+  const updateQueue = props.updateQueue;
+  const profileData = props.optionalProfileDataIfSeparateContext || currentBusiness.profileData;
+  const displayBusinessNameValue = displayBusinessName(profileData);
+  const displayResponsibleOwnerNameValue = displayResponsibleOwnerName(profileData);
+  const taxIdToSubmitToTaxApi = profileData.taxId || "";
+
+  let encryptedTaxIdToSubmitToTaxApi = "";
+  let businessNameToSubmitToTaxApi = "";
+
+  if (profileData.taxId === currentBusiness.profileData.taxId) {
+    encryptedTaxIdToSubmitToTaxApi = profileData.encryptedTaxId || "";
+  }
+
+  if (displayBusinessNameValue) {
+    businessNameToSubmitToTaxApi = profileData.businessName;
+  }
+
+  if (displayResponsibleOwnerNameValue) {
+    businessNameToSubmitToTaxApi = profileData.responsibleOwnerName;
+  }
+
+  if (!businessNameToSubmitToTaxApi) return;
```

Comment:
Looking at the diff between this file and `TaxAccessStepTwo`, why did we need to add this line? It doesn't look like this was something that was originally included.